dan_layer/core/src/workers/hotstuff_waiter.rs

```diff
@@ -425,6 +425,13 @@ where
                 return Ok(());
             }
 
+            // All proposals are for different shards and have the same height, so if one of them is REJECT then there
+            // is no point in continuing to vote.
+            if let Some(reject_node) = proposed_nodes.iter().find(|n| n.justify().decision().is_reject()) {
+                self.abandon_payload(reject_node, &proposed_nodes)?;
```

Comment:
Seems correct. My only concern is that maybe we should vote to our local leader and wait for the local  QC reject before abandoning

Reply:
Thanks I like that better, and then we should always expect the local pledge to exist when abandoning.